appinventor/components-ios/src/Barometer.swift

```diff
@@ -0,0 +1,88 @@
+// -*- mode: swift; swift-mode:basic-offset: 2; -*-
+// Copyright 2023 MIT, All rights reserved
+// Released under the Apache License, Version 2.0
+// http://www.apache.org/licenses/LICENSE-2.0
+
+import Foundation
+import CoreMotion
+
+class Barometer: NonvisibleComponent {
+  private let motionManager = CMAltimeter()
+  private var refreshTimer: Timer?
+  private var enabled = true
+  @objc var AirPressure: NSNumber = 0.0
+  @objc var Available: Bool = false
+  @objc var RefreshTime: TimeInterval = 0.001
+  
+  override init(_ container: ComponentContainer) {
+    super.init(container)
+    checkAvailability()
+    startListening()
+  }
+  
+  @objc var Enabled: Bool {
+    get {
+      return enabled
+    }
+    set(newValue) {
+      if enabled != newValue {
+        enabled = newValue
+        if enabled {
+          startListening()
+        } else {
+          stopListening()
+        }
+      }
+    }
+  }
+  
+  func startListening() {
+    guard Enabled && Available else {
+      return
+    }
+    motionManager.startRelativeAltitudeUpdates(to: OperationQueue.current!) { [weak self] (altitudeData, error) in
+      guard let altitudeData = altitudeData, error == nil else {
+        return
+      }
+      let pressure = self?.calculatePressure(from: altitudeData.relativeAltitude.doubleValue)
+      self?.AirPressure = NSNumber(value: pressure ?? 0.0)
+      if let pressure = pressure {
+        self?.AirPressureChanged(pressure)
+      }
+    }
+    refreshTimer = Timer.scheduledTimer(timeInterval: RefreshTime, target: self, selector: #selector(refreshAirPressure), userInfo: nil, repeats: true)
+  }
+  
+  func stopListening() {
+    motionManager.stopRelativeAltitudeUpdates()
+    refreshTimer?.invalidate()
+    refreshTimer = nil
+  }
+  
+  @objc func checkAvailability() {
+    Available = CMAltimeter.isRelativeAltitudeAvailable()
+  }
+  
+  @objc func refreshAirPressure() {
+    motionManager.startRelativeAltitudeUpdates(to: OperationQueue.current!) { [weak self] (altitudeData, error) in
+      guard let altitudeData = altitudeData, error == nil else {
+        return
+      }
+      let pressure = self?.calculatePressure(from: altitudeData.relativeAltitude.doubleValue)
+      self?.AirPressure = NSNumber(value: pressure ?? 0.0)
+      if let pressure = pressure {
+        self?.AirPressureChanged(pressure)
+      }
+    }
+  }
+  
+  @objc func AirPressureChanged(_ pressure: Double) {
+    EventDispatcher.dispatchEvent(of: self, called: "AirPressureChanged", arguments: NSNumber(floatLiteral: pressure))
```

Comment:
For thread safety, use `DispatchQueue.main.async { ... }` here to ensure the event is dispatched on the main thread.